Gordon360/Services/ProfileService.cs

```diff
@@ -282,6 +283,143 @@ public async Task UpdateProfileLinkAsync(string username, string type, CUSTOM_PR
             await _context.SaveChangesAsync();
         }
 
+        /// <summary>
+        /// convert original fac/staff profile to public fac/staff profile based on individual privacy settings
+        /// </summary>
+        /// <param name="username">username of the fac/staff being searched</param>
+        /// <param name="currentUserType">personnel type of the logged-in user (fac, stu, alu)</param>
+        /// <param name="fac">original profile of the fac/staff being searched</param>
+        /// <returns>public profile of the fac/staff based on individual privacy settings</returns>
+        public PublicFacultyStaffProfileViewModel ToPublicFacultyStaffProfileViewModel
+            (string username, string currentUserType, FacultyStaffProfileViewModel fac)
+        {
+            PublicFacultyStaffProfileViewModel publicFac = (PublicFacultyStaffProfileViewModel)fac;
+            var account = _accountService.GetAccountByUsername(username);
+
+            // select all privacy settings
+            var privacy = _context.UserPrivacy_Settings.Where(up_s => up_s.gordon_id == account.GordonID);
+            // get type of viewmodel for reflection property setting
+            Type fs_vm = new PublicFacultyStaffProfileViewModel().GetType();
+
+            foreach (UserPrivacy_Fields field in _context.UserPrivacy_Fields)
+            {
+                var row = privacy.FirstOrDefault(p => p.Field == field.Field);
+                if (row is UserPrivacy_Settings instance)
+                {
+                    if (instance.Visibility == "Private" || (instance.Visibility == "FacStaff" && currentUserType != "fac"))
+                    {
+                        fs_vm.GetProperty(field.Field).SetValue(publicFac, "Private as requested.");
+                    }
+                }
+                else
+                {
+                    fs_vm.GetProperty(field.Field).SetValue(publicFac, "Private as requested.");
+                }
+            }
+
+            return publicFac;
+        }
+
+        /// <summary>
+        /// convert original student profile to public student profile based on individual privacy settings
+        /// </summary>
+        /// <param name="username">username of the student being searched</param>
+        /// <param name="currentUserType">personnel type of the logged-in user</param>
+        /// <param name="stu">original profile of the student being searched</param>
+        /// <returns>public profile of the student based on individual privacy settings</returns>
+        public PublicStudentProfileViewModel ToPublicStudentProfileViewModel
+            (string username, string currentUserType, StudentProfileViewModel stu)
```

Comment:
```suggestion
        /// <returns>public profile of the student based on individual privacy settings</returns>
        public PublicStudentProfileViewModel ToPublicStudentProfileViewModel
            (string username, string currentUserType, StudentProfileViewModel stu)
```
student privacy is not a part of this PR. This PR is meant to handle fac staff privacy

Reply:
Either